lib/backpex/live_resource.ex

```diff
@@ -10,6 +10,82 @@ defmodule Backpex.LiveResource do
   > When you `use Backpex.LiveResource`, the `Backpex.LiveResource` module will set `@behavior Backpex.LiveResource`. Additionally it will create a LiveView based on the given configuration in order to create fully functional index, show, new and edit views for a resource. It will also insert fallback functions that can be overridden.
   '''
 
+  @options_schema [
+    adapter: [
+      doc: "The datalayer adapter to use.",
+      type: :atom,
+      default: Backpex.Adapters.Ecto
+    ],
+    adapter_config: [
+      doc: "The configuration for the datalayer. See corresponding adapter for possible configuration values.",
+      type: :keyword_list,
+      required: true
+    ],
+    primary_key: [
+      doc: "The primary key used for identifying items.",
+      type: :atom,
+      default: :id
+    ],
+    layout: [
+      doc: "Layout to be used by the LiveResource.",
+      type: :mod_arg,
+      required: true
+    ],
+    pubsub: [
+      doc: "PubSub name of the project.",
+      type: :atom,
+      required: true
+    ],
+    topic: [
+      doc: "The topic for PubSub.",
+      type: :string,
+      required: true
+    ],
+    event_prefix: [
+      doc:
+        "The event prefix for Pubsub, to differentiate between events of different resources when subscribed to multiple resources.",
+      type: :string,
+      required: true
+    ],
+    per_page_options: [
+      doc: "The page size numbers you can choose from.",
+      type: {:list, :integer},
+      default: [15, 50, 100]
+    ],
+    per_page_default: [
+      doc: "The default page size number.",
+      type: :integer,
+      default: 15
+    ],
+    init_order: [
+      doc: "Order that will be used when no other order options are given.",
+      type: {
+        :or,
+        [
+          {:fun, 1},
+          map: [
+            by: [
+              doc: "The column used for ordering.",
+              type: :atom
+            ],
+            direction: [
+              doc: "The order direction",
+              type: :atom
+            ]
+          ]
+        ]
+      },
+      default: Macro.escape(%{by: :id, direction: :asc})
```

Comment:
Just an idea: Make `init_order` required and remove the default because we also support primary keys that are not named `id`